web_services/foxx/brightid/index.js

```diff
@@ -194,156 +196,91 @@ const handlers = {
     });
   },
 
-  allVerificationsGet: function(req, res){
+  verificationPublicGet: function(req, res){
     const appKey = req.param('app');
-    const count_only = 'count_only' in req.queryParams;
     const app = db.getApp(appKey);
-    const context = db.getContext(app.context);
-    const coll = arango._collection(context.collection);
-    let contextIds = db.getLastContextIds(coll, app._key);
-    let data = {
-      count: contextIds.length
-    }
-    if (! count_only){
-      data['contextIds'] = contextIds
+    const roundedTimestamp = req.param('roundedTimestamp');
+    const verification = req.param('verification') || app.verification;
+
+    const tp = app.timestampPrecision;
+    const serverRoundedTimestamp = tp ? parseInt(Date.now() / tp) * tp : 0;
+    if (serverRoundedTimestamp != roundedTimestamp) {
+      throw new errors.InvalidRoundedTimestampError(serverRoundedTimestamp, roundedTimestamp);
     }
+
+    const info = stringify({ app: appKey, roundedTimestamp, verification });
+    const server = new WISchnorrServer();
+    const params = server.GenerateWISchnorrParams(info);
+    cachedParamsColl.insert({
+      _key: params.public,
+      private: params.private,
+      app: appKey,
+      roundedTimestamp,
+      verification,
```

Comment:
I don't think needs to be included.  I think we just need to save the private params keyed by the public params to be able to complete the 3rd step in the interaction.  I assume the creationDate is used to create an index that automatically deletes expired parameters--good idea; if that happens to a client, they can just start over.